doc/dev_guide/c_style_guide_new.rst

```diff
@@ -0,0 +1,1046 @@
+are placed substantially to the right. The same applies to function headers
+with a long argument list. However, never break user-visible strings such as
+printk messages, because that breaks the ability to grep for them.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Chapter 3: Placing Braces and Spaces
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The other issue that always comes up in C styling is the placement of
+braces. Unlike the indent size, there are few technical reasons to
+choose one placement strategy over the other, but the preferred way, as
+shown to us by the prophets Kernighan and Ritchie, is to put the opening
+brace last on the line, and put the closing brace first, thus:
+
+.. code-block:: c
+
+    if (x is true) {
+      we do y
+    }
+
+This applies to all non-function statement blocks (if, switch, for,
+while, do).  E.g.:
+
+.. code-block:: c
+
+    switch (action) {
+    case KOBJ_ADD:
+      return "add";
+    case KOBJ_REMOVE:
+      return "remove";
+    case KOBJ_CHANGE:
+      return "change";
+    default:
+      return NULL;
+    }
+
+However, there is one special case, namely functions: they have the
+opening brace at the beginning of the next line, thus:
+
+.. code-block:: c
+
+    int
+    function(int x)
+    {
+      body of function
+    }
+
+Heretic people all over the world have claimed that this inconsistency
+is ...  well ...  inconsistent, but all right-thinking people know that
+(a) K&R are **right** and (b) K&R are right.  Besides, functions are
+special anyway (you can't nest them in C).
+
+Note that the closing brace is empty on a line of its own, **except** in
+the cases where it is followed by a continuation of the same statement,
+ie a ``while`` in a do-statement or an ``else`` in an if-statement, like
+this:
+
+.. code-block:: c
+
+    do {
+      body of do-loop
+    } while (condition);
+
+and
+
+.. code-block:: c
+
+    if (x == y) {
+      ..
+    } else if (x > y) {
+      ...
+    } else {
+      ....
+    }
+
+Rationale: K&R.
+
+Also, note that this brace-placement also minimizes the number of empty
+(or almost empty) lines, without any loss of readability.  Thus, as the
+supply of new-lines on your screen is not a renewable resource (think
+25-line terminal screens here), you have more empty lines to put
+comments on.
+
+Do not unnecessarily use braces where a single statement will do.
+
+.. code-block:: c
+
+    if (condition)
+      action();
+    and
+
+.. code-block:: none
+
+    if (condition)
+      do_this();
+    else
+      do_that();
+
+This does not apply if only one branch of a conditional statement is a single
+statement; in the latter case use braces in both branches:
+
+.. code-block:: c
+
+    if (condition) {
+      do_this();
+      do_that();
+    } else {
+      otherwise();
+    }
+
+**********************
+Chapter 3.1: Spaces
+**********************
+
+Tarantool style for use of spaces depends (mostly) on
```

Comment:
Does not look like a direct cite. And is not different from Linux. Should we even change anything here? I am talking about using "Tarantool" word in the kernel style guide copy-paste.